### PR TITLE
Feature: API Pane Setting section redesign

### PR DESCRIPTION
**Pull request by devrk96**

### CODE REVIEW

File: app/client/src/pages/Editor/ActionSettings.tsx [hunk 1]
```diff
@@ -25,7 +27,7 @@ const renderEachConfig = (section: any, formName: string): any => {
       try {
         const { configProperty } = formControlOrSection;
         return (
-          <div key={configProperty} style={{ marginTop: "18px" }}>
+          <div key={configProperty} style={{ marginBottom: "25px" }}>
```
Comment (hetunandu) [inline app/client/src/pages/Editor/ActionSettings.tsx:30]: Make this a styled component

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/components/ads/TextInput.tsx [hunk 2]
```diff
@@ -39,10 +39,11 @@ export function notEmptyValidator(value: string) {
 export type TextInputProps = CommonComponentProps & {
   placeholder?: string;
   fill?: boolean;
-  defaultValue?: string;
+  defaultValue?: any;
```
[the remark this answers is not in the document]
Comment (devrk96) [reply app/client/src/components/ads/TextInput.tsx:42]: Using a union type here is causing the type to change everywhere it is used as a text input like for the onChange method or validation functions. It is causing many conflicts like "number can't be defined for a name or email ".

File: app/client/src/pages/Editor/ActionSettings.tsx (hunk 1 above)
Comment (devrk96) [reply app/client/src/pages/Editor/ActionSettings.tsx]: Resolved

File: app/client/src/components/ads/TextInput.tsx [hunk 3]
```diff
@@ -39,10 +39,11 @@ export function notEmptyValidator(value: string) {
...
   validator?: (value: string) => { isValid: boolean; message: string };
-  onChange?: (value: string) => void;
+  onChange?: (value: any) => void;
```
[the remark this answers is not in the document]
Comment (devrk96) [reply app/client/src/components/ads/TextInput.tsx]: Resolved

**Pull request merged**